nimble/host/src/ble_l2cap_coc.c

```diff
@@ -88,17 +103,81 @@ ble_l2cap_coc_create_server(uint16_t psm, uint16_t mtu,
     return 0;
 }
 
-static uint16_t
-ble_l2cap_coc_get_cid(void)
+static int
+ble_l2cap_coc_alloc_cid(cid_mask_t *cid_chan_mask)
+{
+    int i, j;
+    /* The following equation is used to get the Maximum Value of bitmask
+     * cid_mask = ((1UL << TOTAL_CID_PER_MASK_ELEMENT) - 1)
+     * but using -1 is more generic in terms of cid_mask_t as all octets
+     * must be 0xFF
+     * cid_mask is used to search for the first free slot to use it.
+     * */
+    cid_mask_t cid_mask = -1;
+    uint16_t cid_idx;
+
+    for(i = 0; i < CID_MASK_LEN; i++) {
+        if(cid_chan_mask[i] != cid_mask) {
+             break;
+        }
+    }
+
+    if (i == CID_MASK_LEN) {
+        /* No Free CID */
+        return -1;
+    }
+
+    cid_mask = cid_chan_mask[i];
+
+    for(j = 0; j < TOTAL_CID_PER_MASK_ELEMENT; j++ ) {
```

Comment:
Do we want to be strict about number of CoC assigned per ACL?
If so then this function should check it as well.

Reply:
added